### ISSUE TITLE
Please reconsider enabling GitHub Discussions on this repo

### ISSUE THREAD

**Issue by NSExceptional** [microsoft/vscode#125609]
It's been almost a year since #101818, and I'm hoping I can change someone's mind here and get this feature turned on :)

I've read about the various [feedback channels](https://github.com/microsoft/vscode/wiki/Feedback-Channels) available today, and while I think they're all great, they have their downsides:

- Chatrooms like Slack/Gitter are not indexed by Google, which means someone can only see any potential answers to their question by searching both Google and the chatrooms
- Chatrooms involve what I deem to be unnecessary friction:
    - Signing up for an account, giving out your email to a new service
    - Checking your email to verify your account
    - Actually joining the chat
    - Potentially learning how to use a new platform
    
    These could all discourage users from asking for help at all. Users with anxiety are more prone to such discouragement; just the thought of needing to go through so many extra steps to ask a simple question is enough to trigger a minor wave of anxiety and stop someone in their tracks.
- StackOverflow is great for general programming questions imo, but I think specific forums like subreddits and issues/discussions on GitHub are far superior in terms of SEO and overall ease of use since they act as centralized hubs for knowledge related to the topic at hand, and aren't aren't bloated with many of the roadblocks StackOverflow has.
   - Searching for "how to show specific line of text document" on Google or StackOverflow will yield inferior results compared to searching the issues and discussions within this repo.
   - Users who are not active on StackOverflow often cannot even answer questions due to the karma system; a roadblock that is probably not necessary for any of our communities.

Pros of enabling Discussions:
- Search engines can index + better SEO compared to StackOverflow since posts are tied to the repo
- An account is not needed to view questions and answers; greatly reduces friction
- If you do want to ask a question, all you need is a GitHub account, which you probably have
- Provides an official, centralized place for discussion in closer proximity to everyone in the community

Cons of enabling Discussions;
- Adds "one more place" to look for answers (could also be a pro?)
- Might impose extra moderation needs on the team

---

I see in the previous issue it was said that you'd like to try Discussions out on a smaller project first. Has that happened yet?

**Comment by macintacos** [comment]
Just to give my $0.02 on this: there have been a couple of ideas that I've had over the past few months that I have not put into a proper GitHub issue in this repo, mostly because:

1. It's difficult for one to put together a decent "proposal" of sorts, before an issue is opened, that doesn't have flaws that can immediately be picked at (by members of the commmunity or VSCode Engineers themselves), and... 
2. Getting any amount of traction on a given issue before it's closed by the bot for not having enough votes is actually more difficult than it has to be - 20 votes IMO should be a pretty low bar, but you'd be surprised how much self-evangelizing you need to do, outside of "proper" channels, in order to get to that amount, and...
3. It's not easy for folks to naturally find issues that they're interested in based on the GitHub Issues interface, because most of the labels/categories are geared towards develop work rather than user engagement.

I feel like Discussions (or just something more permanent/"formal" than Slack, less "please help me" than Stack Overflow) would have the following additional benefits that aren't touched on in the initial description:

- It'd be a good place to flesh out ideas, garner traction, and then present them in issues that
	1. ...are more likely to not immediately fall away into obscurity because users don't bother to patrol the GitHub Issues list (which is pretty noisy) and...
	2. ...allow well-thought proposals to come through in GitHub issues that _are_ eventually filed, and thus not waste VSCode Engineering time trying to poke holes in ideas that'd be hopefully be sowed up by the time an issue actually is opened.
- It'd allow the VSCode team to categorize things into meaningful "buckets" for discussion from the community like allowing folks to show off their configs, their extensions in progress, their ideas before they file actual "issues", provide "general" feedback, etc. - [github/feedback](https://github.com/github/feedback/discussions) is a nice model to go by.

I've had really good experiences with community forums powered things like by Discourse ([Swift's forums](https://forums.swift.org/), [Obsidian's forums](https://forums.obsidian.net/), [Draft's forums](https://forums.getdrafts.com/), etc. etc.), and I've always been surprised that the VSCode team doesn't have something similar. Considering that Discussions is developed by a company that shares the same parent as VSCode, the synergy there feels obvious and natural, so hopefully it's something y'all would seriously consider 🙂.

If the issue is about spreading yourselves too thin, I'd even say that Discussions could be a replacement for things like Stack Overflow, if given enough time to bake, and would encourage users to interact with things like announcements, feature additions, etc. For example, I would've loved to just voice my excitement about a few things in the most recent release of VSCode, but it seems like there's no official channel to even discuss them (not even on [the official changelog place itself](https://code.visualstudio.com/updates/v1_57)), which feels like a shame.

Anyway, apologies for rambling a bit - hope the above makes sense.

**Comment by sharpninja** [comment]
Please enable discussions.  Half the time I don't say anything when I have a question because I don't want to clutter the issues with what may or may not be trivial, and if I don't ask, I don't know whether it's trivial or not.

For example, I don't know if this warrants an issue:

How do you use null coalescing in settings?  

```json
{
    "powershell.cwd": "${fileDirname}"
}
```

If `$fileDirename` exists, the setting is fine, but if it's undefined then the powershell extension fails to start.  It seems reasonable that this should work:

```json
{
    "powershell.cwd": "${fileDirname ?? workspaceFolder}"
}
```

And that results in:
![image](https://user-images.githubusercontent.com/16146732/122417869-6d6a9680-cf4f-11eb-8071-2e29918e59c4.png)

**Comment by NSExceptional** [comment]
As an example of why StackOverflow is generally bad for this, [this question](https://stackoverflow.com/questions/47888617/why-textdocumentcontentprovider-dont-call-providetextdocumentcontent-on-update-w) was asked over 3 years ago and got no attention at all, despite having nearly 300 views. I'm positive it would have gotten an answer on the Discussion page here.

**Comment by NSExceptional** [comment]
@egamma Can I get your thoughts on this? 🙏🏻

**Comment by xaxazak** [comment]
I came here to add this as an issue, so I'll just add an example and some reasoning.

The example:
I wanted to discuss the right way to add an option to VSCode settings. I have the functionality working. I could spend time following through other options to see how they're done but I can't be sure I'd do it 100% properly, which would be important if I want to submit it as a patch.

What is the right place to discuss this? It's too back-and-forth for StackOverflow, it's not likely to yield results if I submit my question as an issue, and it'll just get buried over time in Slack or other continuous-discussion-stream sites.


Forums work by having a useful heirarchy: Forum -> [Subforum] -> Topic -> Post. The Topic bit is extremely useful for increasing visibility of specific issues while reducing the information load, which helps helpers make far more efficient use of their time - rather than having to scroll past every post about an issue they can't help with, such as on slack. Forums have topic-reply counts and topics can be marked solved, further reducing information load. And, as said above, forums are also indexed better by search engines. 

In my view, dropping slack and/or gitter would be the better option. StackOverflow can't really be "dropped", but it doesn't cover many query types, such as subjective ones or ones where back-and-forth discussion is necessary.

**Comment by NSExceptional** [comment]
Looks like this closes this issue and #113561

https://github.com/microsoft/vscode-discussions
